### PR TITLE
[FLINK-20239][docs] Confusing pages: Hive Read & Write and Hive Strea… 

### PR DESCRIPTION
**Pull request by sjwiesman**
## What is the purpose of the change

Consolidate "Hive Read & Write" and "Hive Streaming" to provide users a unified overview of reading and writing to and from hive. 

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

### CODE REVIEW

**Comment by sjwiesman** [comment]
@dawidwys and @lirui-apache would either of you like to take a look at this?

File: docs/dev/table/hive/hive_read_write.md [hunk 1]
```diff
@@ -22,119 +22,199 @@ specific language governing permissions and limitations
...
+1) The Hive catalog must be set as the current catalog before you can query the view. 
+This can be done by either `tableEnv.useCatalog(...)` in Table API or `USE CATALOG ...` in SQL Client.
 
-# ------ See all registered database in catalog 'mytable' ------
+2) Hive and Flink SQL have different syntax, e.g. different reserved keywords and literals.
+Make sure the view’s query is compatible with Flink grammar.
 
-Flink SQL> show databases;
-default
+### Temporal Table Join
```
Comment (lirui-apache) [inline docs/dev/table/hive/hive_read_write.md:114]: I wonder whether we should list all the source tables that can be used as temporal tables (maybe in [this page](https://ci.apache.org/projects/flink/flink-docs-release-1.11/dev/table/streaming/temporal_tables.html#temporal-table)). And then we can put a cross link here. It's annoying for users to have to check each connector doc to decide whether a table can be used in temporal join. And maybe users don't even expect to find such information in the "read/write" page.

File: docs/dev/table/hive/hive_read_write.md [hunk 2]
```diff
@@ -22,119 +22,199 @@ specific language governing permissions and limitations
...
+{% highlight sql %}
 # ------ INSERT INTO will append to the table or partition, keeping the existing data intact ------ 
 Flink SQL> INSERT INTO mytable SELECT 'Tom', 25;
 
 # ------ INSERT OVERWRITE will overwrite any existing data in the table or partition ------ 
 Flink SQL> INSERT OVERWRITE mytable SELECT 'Tom', 25;
 {% endhighlight %}
 
-We support partitioned table too, Consider there is a partitioned table named myparttable with four columns: name, age, my_type and my_date, in types ...... my_type and my_date are the partition keys.
+Data can also be inserted into a particular partitions.
```
Comment (lirui-apache) [inline docs/dev/table/hive/hive_read_write.md:215]:
```suggestion
Data can also be inserted into particular partitions. 
```

**Review by lirui-apache — COMMENTED** [review]
Thanks @sjwiesman for taking the efforts to fix the issue! I think the consolidation makes the document clearer. LGTM overall with two minor comments.

File: docs/dev/table/hive/hive_read_write.md (hunk 1 above)
Comment (sjwiesman) [reply docs/dev/table/hive/hive_read_write.md:114]: There's an open PR to expand temporal table join documentation, why don't you comment there #14152

**Comment by sjwiesman** [comment]
Thanks for the review, merging . . .

**Pull request closed without merging**